src/Workspaces/CSharp/Portable/Simplification/Reducers/AbstractCSharpReducer.AbstractReductionRewriter.cs

```diff
@@ -67,28 +68,31 @@ public void RequireInitialized()
                 Contract.ThrowIfNull(SemanticModel);
             }
 
-            private static SyntaxNode? GetParentNode(SyntaxNode node)
+            private static SyntaxNode GetParentNode(SyntaxNode node)
                 => node switch
                 {
                     ExpressionSyntax expression => GetParentNode(expression),
                     PatternSyntax pattern => GetParentNode(pattern),
                     CrefSyntax cref => GetParentNode(cref),
-                    _ => null
+                    _ => node.GetRequiredParent(),
```

Comment:
the deleted SimplifyExpression method had a null check for parent node, was that unnecessary? Otherwise, this change would cause us to throw.